xknxproject/models/models.py

```diff
@@ -30,15 +32,24 @@ def __init__(
         self.project_uid = project_uid
         self.description = description
         self.dpt = dpt
+        self.comment = comment
+        self.style = style
+        self.address = XMLGroupAddress.str_address(self.raw_address, self.style)
 
-        self.address = self._parse_address()
-
-    def _parse_address(self) -> str:
+    @staticmethod
+    def str_address(raw_address: int, group_address_style: GroupAddressStyle) -> str:
         """Parse a given address and returns a string representation of it."""
-        main = (self.raw_address & 0b1111100000000000) >> 11
-        middle = (self.raw_address & 0b11100000000) >> 8
-        sub = self.raw_address & 0b11111111
-        return f"{main}/{middle}/{sub}"
+        if group_address_style == GroupAddressStyle.FREE:
+            return str(raw_address)
+        main = (raw_address & 0b1111100000000000) >> 11
+        if group_address_style == GroupAddressStyle.THREELEVEL:
+            middle = (raw_address & 0b11100000000) >> 8
+            sub = raw_address & 0b11111111
+            return f"{main}/{middle}/{sub}"
+        if group_address_style == GroupAddressStyle.TWOLEVEL:
+            sub = raw_address & 0b11111111111
+            return f"{main}/{sub}"
+        raise ValueError(f"GroupAddressSyste '{group_address_style}' not supported!")
```

Comment:
In `XMLGroupRange.str_address()` we have no fall-through handling. Mypy seems happy with that - so maybe it would be happy here too. I wonder why it doesn't complain about "unreachable code" though...
If you want, you could try to remove this line and see if mypy is still happy.